py/parse.c

```diff
@@ -427,6 +417,25 @@ STATIC mp_obj_t mp_parse_node_convert_to_obj(mp_parse_node_t pn) {
 }
 #endif
 
+STATIC bool parse_node_is_const_bool(mp_parse_node_t pn, bool value) {
+    // Returns true if 'pn' is a constant whose boolean value is equivalent to 'value'
+    #if MICROPY_COMP_CONST_TUPLE || MICROPY_COMP_CONST
+    return mp_parse_node_is_const(pn) && (bool)mp_obj_is_true(mp_parse_node_convert_to_obj(pn)) == value;
+    #else
+    // stdbool cast is needed here to convert int to exactly true/false
+    return MP_PARSE_NODE_IS_TOKEN_KIND(pn, value ? MP_TOKEN_KW_TRUE : MP_TOKEN_KW_FALSE)
+           || (MP_PARSE_NODE_IS_SMALL_INT(pn) && (bool)MP_PARSE_NODE_LEAF_SMALL_INT(pn) == value);
+    #endif
+}
+
+bool mp_parse_node_is_const_false(mp_parse_node_t pn) {
```

Comment:
would code size be reduced further by making these two functions `static inline` in `py/parse.h`?

Reply:
Good suggestion, I hadn't checked. Turns out this undoes the 16 byte size saving, though!